### ISSUE TITLE
Ability to deprecate aliases

### ISSUE THREAD

**Issue by greg0ire** [symfony/symfony#24507]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.10

I'm getting deprecations from bundles such as described [here](https://github.com/php-tmdb/symfony/issues/24), and it seems the solution to those deprecations would be to rename services so that the id matches the FQCN, and then alias those services to the previous service id for BC.
I thought deprecating the aliases would be a good idea too, but when I try, it looks like this is not supported, and I get the following error:

> Using the element "deprecated" is deprecated for the service "nelmio_alice.faker.generator" which is defined as an alias in "/var/www/html/vendor/nelmio/alice/src/Bridge/Symfony/DependencyInjection/../Resources/config/faker.xml". The XmlFileLoader will raise an exception in Symfony 4.0, instead of silently ignoring unsupported elements.

Here is what I tried:

```xml
        <service id="nelmio_alice.faker.generator" alias="Faker\Generator">
            <deprecated>The "nelmio_alice.faker.generator" is deprecated in favor of the Faker\Generator FQCN.</deprecated>
        </service>
        <service id="Faker\Generator"
                 class="Faker\Generator">
            <factory class="Faker\Factory" method="create" />

            <argument>%nelmio_alice.locale%</argument>

            <call method="seed">
                <argument>%nelmio_alice.seed%</argument>
            </call>

            <!-- Calls to add tagged providers are made in a compiler pass -->
        </service>

```

I understand the `deprecated` element I tried to use in my alias is not supported. Should Symfony provide the ability to deprecate aliases, or is there a better way to avoid the deprecations I'm talking about?

**Comment by theofidry** [comment]
👍  I'm up for it. An argument to avoid extra arguments on an alias is that an alias is only meant to point to another service and not provide other extra behaviours. However an alias is still a service ID and as such you might want to deprecate it [that specific service ID, not the underlying service].

A perfect real life examples is the except given above by @greg0ire for [alice](https://github.com/nelmio/alice).

**Comment by j92** [comment]
I have been working on implementing the deprecation of aliases. I applied the same logic as for services, so we can deprecate an alias like this: 
```xml
<service id="alias_for_foobar" alias="foobar">
      <deprecated />
</service>
```

Or we can deprecate an alias providing a deprecation template:
 ```xml
<service id="alias_for_foobar" alias="foobar">
      <deprecated>The "%service_id%" service is deprecated.</deprecated>
</service>
```

I submitted a WIP pull request to get early feedback: #24707.

**Comment by nicolas-grekas** [comment]
Time to deprecate some aliases, PR welcome :)

**Comment by javiereguiluz** [comment]
Closing as fixed by #29968.
